scripts/generate_header.py

```diff
@@ -155,6 +155,21 @@ class DeviceBase:
 
 # sub templates
 # generate sub parts of template
+
+def integer_from_python_to_c(an_int: int) -> str:
+    """
+    Takes a positive python integer and return a string that can
+    be used as a constant in C. This means that a L will be prepended
+    if the value is greater than 32 bits.
+
+    :param an_int: an integer value to be written to a c file
+    :return: a string that can be used as a c constant
+    """
+    if an_int > 0xFFFFFFFF:
+        return hex(an_int) + 'ULL'
```

Comment:
Would it be harmful to just always append the `ULL` suffix to the integer literals, even if it's smaller than 32 bits? That may simplify the logic (by obviating most of it).

Also, if you _don't_ unconditionally add the `ULL`, can you at least always add the `U` in the < 2^32 case? It would be really confusing to have it be signed in the case that it is smaller than 2^32, especially since that is a much more likely case.

Reply:
It is good practice, but doesn't fix the error. I did fix that by changing a u32_t to a uintptr_t in block-pio-sifive/drivers/metal/sifive_pio.c

Sloppiness on my part, assuming that pio was always 32 bit.